src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTracePipelineLogic.java

```diff
@@ -180,63 +180,63 @@ private void updateBuildData(BuildData buildData, Run<?, ?> run, BuildPipelineNo
         }
     }
 
-
     private void sendTrace(final Tracer tracer, final BuildData buildData, final BuildPipelineNode current, final SpanContext parentSpanContext) {
         if(!isTraceable(current)){
-            logger.severe("Node " + current.getName() + " is not traceable.");
-            return;
-        }
-
-        final Tracer.SpanBuilder spanBuilder = tracer.buildSpan(buildOperationName(current)).withStartTimestamp(current.getStartTimeMicros());
-
-        if(parentSpanContext != null) {
-            spanBuilder.asChildOf(parentSpanContext);
-        }
-
-        spanBuilder
-                .withTag(DDTags.SERVICE_NAME, DatadogUtilities.getDatadogGlobalDescriptor().getTraceServiceName())
-                .withTag(DDTags.RESOURCE_NAME, current.getName())
-                .withTag(DDTags.SPAN_TYPE, "ci")
-                .withTag(DDTags.LANGUAGE_TAG_KEY, "");
-
-        final Map<String, Object> traceTags = buildTraceTags(current, buildData);
+            for(final BuildPipelineNode child : current.getChildren()) {
+                sendTrace(tracer, buildData, child, parentSpanContext);
+            }
+        } else {
```

Comment:
nit: you could you return after the for loop instead of this big else block?

Reply:
Changed! 👌🏼